### PR TITLE
Improve error message in Kubelet CPU assignment logic

### PR DESCRIPTION
**Pull request by matte21**
Include number of requested and available CPUs in the error message when the assignment of CPUs fails because there are less available CPUs than requested.

#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
Include number of requested and available CPUs in the error message when the assignment of CPUs fails because there are less available CPUs than requested.

The new error message should speed up troubleshooting.

#### Special notes for your reviewer:

This change might be considered breaking, depending on whether the wording of error messages is guaranteed to be stable. i.e. if some client code examines the error by inspecting its message (e.g. via regexp), that client might break. The old wording was locked by unit tests, so I had to change some expected unit tests results.

#### Does this PR introduce a user-facing change?

```release-note
When the Kubelet fails to assign CPUs to a Pod because there less available CPUs than the Pod requests, the error message changed from
"not enough cpus available to satisfy request" to "not enough cpus available to satisfy request: <num_requested> requested, only <num_available> available".
```

### CODE REVIEW

File: pkg/kubelet/cm/cpumanager/cpu_assignment.go [hunk 1]
```diff
@@ -453,7 +453,7 @@ func takeByTopologyNUMAPacked(topo *topology.CPUTopology, availableCPUs cpuset.C
 		return acc.result, nil
 	}
 	if acc.isFailed() {
-		return cpuset.New(), fmt.Errorf("not enough cpus available to satisfy request")
+		return cpuset.New(), fmt.Errorf("not enough cpus available to satisfy request: %d requested, only %d available", numCPUs, availableCPUs.Size())
```
Comment (ffromani) [inline pkg/kubelet/cm/cpumanager/cpu_assignment.go:456]: I'd go for a terser form: "requested %d, available %d" or even "requested=%d, available=%d". The current wording yeilds a nicer sentence, but IMO here practicality beats nice. Applies to everything below.

**Review by ffromani — COMMENTED** [review]
It's (still) unclear if we consider error messages part of the API contract. They probably are not, but the observability in this area is historically poor, so it's likely there's code/tooling (including tests) actually depending on the messages. Appending to the existing text is probably the best we can do.

There's an effort aiming to improve this aspect, stemming from the effort of GA'ing memory manager but pertaining to all resource managers (obv. incl. CPU manager). Let's see how this PR can fit in this picture.

**Comment by matte21** [comment]
The force push to [d4a5a08](https://github.com/kubernetes/kubernetes/pull/121059/commits/d4a5a085a84b3d2c3853b589787e25faec33e380) rebases on main and addresses the review comments.

File: pkg/kubelet/cm/cpumanager/cpu_assignment.go (hunk 1 above)
Comment (matte21) [reply pkg/kubelet/cm/cpumanager/cpu_assignment.go]: Done in [d4a5a08](https://github.com/kubernetes/kubernetes/pull/121059/commits/d4a5a085a84b3d2c3853b589787e25faec33e380)

**Comment by matte21** [comment]
@ffromani

**Comment by ffromani** [comment]
/lgtm

I think it's a nice little improvement

**Comment by derekwaynecarr** [comment]
it is fine to update error messages.

/approve

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**